### ISSUE TITLE
[All connectors] Define which entity types a connector is allowed to create or update

### ISSUE THREAD

**Issue by amr-cossi** [OpenCTI-Platform/opencti#1025]
## Problem to Solve

For a better compatibility between connectors ingesting data, it would be useful to have granular settings to choose which external database has the precedence for each type of entity. For example, if you decide that your TTPs reference is handled by MITRE ATT&CK, other connectors shouldn't be able to override TTP entities with their own data, only create links with them.
To disallow the creation of one entity type seems more constrained as it would loose part of the data you are importing but should still be a setting, as one could restrict one entity type to a finite number of possible values and discard other data.

## Current Workaround

Not activate connectors that pollute the reference dataset or accept the pollution, then cleaning manually.

## Proposed Solution

For each connector, adding the possibility to restrict entity types that can be created or updated by this connector.
This could be handled in the core platform by being able to set more granular rights on any user, including the connector token, or at the connectors level with compulsory config for all.

## Additional Information

The more granular version would be to do the same at the property level (title, description of an entity).
I didn't put much thought on the same feature regarding relationships (restrict by source type, destination type and role) but it could be also another possibility of a more granular feature.

**Comment by SamuelHassine** [comment]
@amr-cossi: This is true. The parameter `UPDATE_EXISTING_DATA` will be used to know if an automatic merge should occurred or not. Your explanation is pretty clear. The behavior will be exactly as you described:

If `UPDATE_EXISTING_DATA` is set to `true` in a connector and this connector produces an entity which resolves multiple existing entities in the platform, then all these entities will be merged together, taking as a reference the entity the connector tries to create.

Is this ok for you?

**Comment by amr-cossi** [comment]
As a first improvement I'd say yes :

- I don't see a case here I would want an auto merge but no update of existing fields. The new entity added is obviously the reference.
- The other way around, if a connector doesn't have the right to update entities, it doesn't have the right to merge anything either.

The issue will remain open for the more granular feature : being able to change this parameter based on entity type.
